src/api/admin.ts

```diff
@@ -0,0 +1,61 @@
+import firebase from "firebase/app";
+import { UserInfo } from "firebase";
+import _ from "lodash";
+
+export interface VenueInput {
+  name: string;
+  bannerImageFile: FileList;
+  logoImageFile: FileList;
+  tagline: string;
+  longDescription: string;
+}
+
+type FirestoreVenueInput = Omit<
+  VenueInput,
+  "bannerImageFile" | "logoImageFile"
+> & {
+  bannerImageUrl: string;
+  logoImageUrl: string;
+};
+
+export const createUrlSafeName = (name: string) => name.replace(/\W/g, "");
+
+export const createVenue = async (input: VenueInput, user: UserInfo) => {
+  const storageRef = firebase.storage().ref();
+
+  const logoFile = input.logoImageFile[0];
+  const bannerFile = input.logoImageFile[0];
+
+  const urlVenueName = createUrlSafeName(input.name);
+  console.log("createVenue -> user", user);
+
+  // upload logo file
+  const uploadLogoRef = storageRef.child(
+    `users/${user.uid}/venues/${createUrlSafeName(urlVenueName)}/${
```

Comment:
You're calling `createUrlSafeName` the result already

Reply:
what do you mean?

Reply:
const urlVenueName = createUrlSafeName(input.name); ([l29](https://github.com/sparkletown/sparkle/pull/253/files#diff-7b49daef5f76b8f0c9a29c54f8c5e376R29))
... 
... createUrlSafeName(urlVenueName)

Reply:
ah yes.. fix incoming